src/core/io/mpiio/mpiio.cpp

```diff
@@ -446,9 +446,11 @@ void mpi_mpiio_common_read(const char *filename, unsigned fields) {
 
     for (int i = 0; i < nlocalpart; ++i) {
       int blen = boff[i + 1] - boff[i];
-      auto &bl = local_particles[id[i]]->bl;
-      bl.resize(blen);
-      std::copy_n(&bond[boff[i]], blen, bl.begin());
+      if (blen) {
+        auto &bl = local_particles[id[i]]->bl;
+        bl.resize(blen);
+        std::copy_n(&bond[boff[i]], blen, bl.begin());
```

Comment:
I think it  would be better to use `std::vector::data` instead of taking the address of the first element. That makes the intent clearer, and works also with an empty vector, and hence also does not need the check.

Reply:
I don‘t think `bond.data() + boff[i]` is that much clearer than `&bond[boff[i]].` Most people reading code tend to be confused by pointer arithmetic like this.

Reply:
I don't agree, this is not incidental, but the offset calculation is the actual logic here. Also getting rid of the control structure is worth it. We can also do iterator arithmetic instead if you think that is less scary.

Reply:
using iterator arithmetic now